src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -1920,11 +1921,12 @@ internal PSModuleInfo LoadModuleManifest(
             else if ((requiredProcessorArchitecture != ProcessorArchitecture.None) &&
                      (requiredProcessorArchitecture != ProcessorArchitecture.MSIL))
             {
-                #pragma warning disable SYSLIB0037
-                ProcessorArchitecture currentArchitecture = typeof(object).Assembly.GetName().ProcessorArchitecture;
-                #pragma warning restore SYSLIB0037
+                Architecture currentArchitecture = RuntimeInformation.ProcessArchitecture;
 
-                if (currentArchitecture != requiredProcessorArchitecture)
+                if ((requiredProcessorArchitecture == ProcessorArchitecture.X86 && currentArchitecture != Architecture.X86) ||
+                    (requiredProcessorArchitecture == ProcessorArchitecture.Amd64 && currentArchitecture != Architecture.X64) ||
+                    (requiredProcessorArchitecture == ProcessorArchitecture.Arm && currentArchitecture != Architecture.Arm64) ||
```

Comment:
What if `currentArchitecture` is `Arm` or `Armv6` (no idea what this is :))

Reply:
Module manifest only allows `Arm`, no `Arm64`, and since .NET only supports `Arm64`, `arm == arm64`

Reply:
Hmm, it seems .NET 9 supports ARM32: https://github.com/dotnet/runtime/discussions/71042#discussioncomment-6745103

Reply:
`ProcessorArchitecture` only has `arm`.  So module manifests can't specify arm32 vs arm64.  We could allow for either arm32 or arm64, but while arm64 can run arm32, the opposite isn't true and will fail at runtime.